scripts/src/test/org/oppia/android/scripts/XMLSyntaxCheckTest.kt

```diff
@@ -0,0 +1,155 @@
+package org.oppia.android.scripts
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import org.oppia.android.testing.assertThrows
+import java.io.ByteArrayOutputStream
+import java.io.PrintStream
+
+/** Tests for [XMLSyntaxCheck]. */
+class XMLSyntaxCheckTest {
+  private val outContent: ByteArrayOutputStream = ByteArrayOutputStream()
+  private val originalOut: PrintStream = java.lang.System.out
+
+  @Rule
+  @JvmField
+  public var tempFolder: TemporaryFolder = TemporaryFolder()
+
+  @Before
+  fun setUpTests() {
+    tempFolder.newFolder("testfiles")
+    java.lang.System.setOut(PrintStream(outContent))
+  }
+
+  @After
+  fun restoreStreams() {
+    java.lang.System.setOut(originalOut)
+  }
+
+  @Test
+  fun testXmlSyntax_validXML_xmlSyntaxIsCorrect() {
+    val validXML =
+      "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n" +
+        "<shape xmlns:android=\"http://schemas.android.com/apk/res/android\"\n" +
+        "  android:shape=\"rectangle\">\n" +
+        "  <solid android:color=\"#3333334D\" />\n" +
+        "  <size android:height=\"1dp\" />\n" +
+        "</shape>"
+
+    val tempFile = tempFolder.newFile("testfiles/TestFile.xml")
+    tempFile.writeText(validXML)
+
+    runScript()
+
+    assertThat(outContent.toString().trim()).isEqualTo(
+      ScriptResultConstants.XML_SYNTAX_CHECK_PASSED
+    )
+  }
+
+  @Test
+  fun testXmlSyntax_invalidOpeningTag_xmlSyntaxIsInCorrect() {
+    val invalidXML =
+      "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n" +
+        "<shape xmlns:android=\"http://schemas.android.com/apk/res/android\"\n" +
+        "  android:shape=\"rectangle\">\n" +
+        "  <<solid android:color=\"#3333334D\" />\n" +
+        "  <size android:height=\"1dp\" />\n" +
+        "</shape>"
+    val tempFile = tempFolder.newFile("testfiles/TestFile.xml")
+    tempFile.writeText(invalidXML)
+
+    val exception = assertThrows(Exception::class) {
+      runScript()
+    }
+
+    assertThat(exception).hasMessageThat().contains(
+      ScriptResultConstants.XML_SYNTAX_CHECK_FAILED
+    )
```

Comment:
This can fit on one line--suggest doing that here & elsewhere (including #3340).

Also, maybe directly import the constant to avoid needing to qualify it (though that may be eliminated, anyway, if we stop using the constants object).

Reply:
Done.